File: src/com/betterzhang/learnkotlin/java/ClassZZ.java
package com.betterzhang.learnkotlin.java;

/**
 * Created by IntelliJ IDEA.
 * Author : Andrew Zhang
 * Email  : betterzhang.dev@gmail.com
 * Time   : 2017/06/29 下午 4:57
 * Desc   : 获得class的实例
 */
public class ClassZZ {

    public static void main(String[] args) {
        Class<?> clazz = ClassZZ.class;
        System.out.println(clazz);

        ClassZZ c = new ClassZZ();
        Class<?> clazz2 = c.getClass();
        System.out.println(clazz2);
    }

}

File: src/com/betterzhang/learnkotlin/kotlin/ClassZZ.kt
package com.betterzhang.learnkotlin.kotlin

/**
 * Created by IntelliJ IDEA.
 * Author : Andrew Zhang
 * Email  : betterzhang.dev@gmail.com
 * Time   : 2017/06/29 下午 5:03
 * Desc   : 获得class的实例
 */
class ClassZZ {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            // ClassZZ::class拿到的是Kotlin的KClass,再转换成Java的class实例
            val clazz = ClassZZ::class.java
            println(clazz)

            val c = ClassZZ()
            val clazz2 = c.javaClass
            println(clazz2)
        }
    }

}